code/association_scan/TensorQTL/TensorQTL.ipynb

```diff
@@ -1351,6 +1353,30 @@
     "            print(f\"Output Preview (first 5 rows):\")\n",
     "            print(combined_results.iloc[0:min(5, len(combined_results)), 0:10])\n",
     "        \n",
+    "        if $[pval] > 0:\n",
+    "            # Record initial number of variants\n",
+    "            initial_n = len(combined_results)\n",
+    "\n",
+    "            # Calculate p-value distribution by each 10th percentile (before filtering)\n",
+    "            pval_percentiles = np.percentile(combined_results['pvalue'], np.arange(0, 110, 10))\n",
+    "\n",
+    "            # Filter combined_results by p-value threshold\n",
+    "            combined_results = combined_results[combined_results['pvalue'] < $[pval]]\n",
+    "\n",
+    "            # Print summary\n",
+    "            print(f\"Number of variants initially: {initial_n}\")\n",
+    "            print(f\"Number of variants after filtering: {len(combined_results)}\")\n",
+    "            print(\"P-value distribution by each 10th percentile (before filtering):\")\n",
+    "            print(dict(zip([f\"{i}%\" for i in range(0, 110, 10)], pval_percentiles)))\n",
+    "\n",
+    "            # Save summary to TSV\n",
+    "            summary_df = pd.DataFrame({\n",
+    "                'metric': ['initial_n', 'after_filtering'] + [f\"{i}%\" for i in range(0, 110, 10)],\n",
+    "                'value': [initial_n, len(combined_results)] + list(pval_percentiles)\n",
+    "            })\n",
+    "\n",
+    "            summary_df.to_csv(f\"$[_output['nominal']:nn].summary_stats.tsv\", sep=\"\\t\", index=False)\n",
```

Comment:
In our field `summary stats` has a very specific meaning. To avoid confusion we should call it `summary` not `summary_stats`